samples/attested_tls/common/openssl_utility.cpp

```diff
@@ -0,0 +1,268 @@
+        &private_key_buffer_size);
+
+    if (result != OE_OK)
+    {
+        printf(" failed with %s\n", oe_result_str(result));
+        goto done;
+    }
+
+    printf("public_key_buf_size:[%ld]\n", public_key_buffer_size);
+    printf("public key used:\n[%s]", public_key_buffer);
+
+    result = oe_generate_attestation_certificate(
+        certificate_subject_name,
+        private_key_buffer,
+        private_key_buffer_size,
+        public_key_buffer,
+        public_key_buffer_size,
+        &output_certificate,
+        &output_certificate_size);
+
+    if (result != OE_OK)
+    {
+        printf(" failed with %s\n", oe_result_str(result));
+        goto done;
+    }
+
+    // temporary buffer required as if d2i_x509 call is successful
+    // certificate_buffer_ptr is incremented to the byte following the parsed
+    // data. sending certificate_buffer_ptr as argument will keep
+    // output_certificate pointer undisturbed.
+    certificate_buffer_ptr = output_certificate;
+
+    if ((certificate = d2i_X509(
+             nullptr,
+             &certificate_buffer_ptr,
+             (long)output_certificate_size)) == nullptr)
+    {
+        printf("Failed to convert DER format certificate to X509 structure\n");
+        goto done;
+    }
+    mem = BIO_new_mem_buf((void*)private_key_buffer, -1);
+    if (!mem)
+    {
+        printf("Failed to convert private key buf into BIO_mem\n");
+        goto done;
+    }
+    if ((pkey = PEM_read_bio_PrivateKey(mem, nullptr, 0, nullptr)) == nullptr)
+    {
+        printf("Failed to convert private key buffer into EVP_KEY format\n");
+        goto done;
+    }
+
+    result = OE_OK;
+done:
+    certificate_buffer_ptr = nullptr;
+    BIO_free(mem);
+    oe_free_key(private_key_buffer, private_key_buffer_size, nullptr, 0);
+    oe_free_key(public_key_buffer, public_key_buffer_size, nullptr, 0);
+    oe_free_attestation_certificate(output_certificate);
+    return result;
+}
+
+oe_result_t load_tls_certificates_and_keys(
+    SSL_CTX* ctx,
+    X509*& certificate,
+    EVP_PKEY*& pkey)
+{
+    oe_result_t result = OE_FAILURE;
+    if (generate_certificate_and_pkey(certificate, pkey) != OE_OK)
+    {
+        printf("Cannot generate certificate and pkey\n");
+        goto exit;
+    }
+    if (!SSL_CTX_use_certificate(ctx, certificate))
+    {
+        printf("Cannot load certificate on the server\n");
+        goto exit;
+    }
+
+    if (!SSL_CTX_use_PrivateKey(ctx, pkey))
+    {
+        printf("Cannot load private key on the server\n");
+        goto exit;
+    }
+
+    /* verify private key */
+    if (!SSL_CTX_check_private_key(ctx))
+    {
+        printf("Private key does not match the public certificate\n");
+        goto exit;
+    }
+    result = OE_OK;
+exit:
+    return result;
+}
+
+oe_result_t initalize_ssl_context(SSL_CONF_CTX*& ssl_conf_ctx, SSL_CTX*& ctx)
+{
+    oe_result_t ret = OE_FAILURE;
+    // This is allowed cipher list and curves as per Crypto Board Program
```

Comment:
I would suggest stating it differently; 

// Configure the SSL context based on Open Enclave's security guidance.